controllers/operatorpolicy_controller.go

```diff
@@ -60,166 +65,201 @@ var _ reconcile.Reconciler = &OperatorPolicyReconciler{}
-				OpLog.Error(err, "Could not handle missing musthave object")
-
-				return err
+				panic(err)
 			}
 
-			r.setCompliance(policy, policyv1.Compliant)
+			return reconcile.Result{Requeue: true}, err
+		} else if policy.Spec.OperatorGroup != nil {
+			// If inform mode, keep going and return at the end without requeue. Before
+			// continuing, should check if required og spec is created to set compliance state
 
-			return nil
+			// Set to non compliant because operatorgroup does not exist on cluster, but
+			// should still go to check subscription
+			r.setCompliance(policy, policyv1.NonCompliant)
 		}
+	}
 
-		// Inform
-		r.setCompliance(policy, policyv1.NonCompliant)
-
-		return nil
+	// Retrieve subscription from cache
+	cachedSubscription, err := r.DynamicWatcher.GetFromCache(
+		schema.GroupVersionKind{Group: "operators.coreos.com", Version: "v1alpha1", Kind: "Subscription"},
+		policy.Namespace, policy.Spec.Subscription.SubscriptionSpec.Package)
+	if err != nil {
+		panic(err)
 	}
 
-	// Object exists but it should not exist, delete object
-	// Deleting related objects will be added in the future
-	if exists && !shouldExist {
+	// Create new subscription
+	if cachedSubscription == nil {
 		if strings.EqualFold(string(policy.Spec.RemediationAction), string(policyv1.Enforce)) {
-			OpLog.Info("deleting kind " + subscriptionSpec.Kind + " in ns " + subscriptionSpec.Namespace)
-			err = r.Delete(context.TODO(), subscriptionSpec)
+			subscriptionSpec := buildSubscription(policy, new(operatorv1alpha1.Subscription))
+			OpLog.Info("creating kind " + subscriptionSpec.Kind + " in ns " + subscriptionSpec.Namespace)
 
+			err := r.Create(context.TODO(), subscriptionSpec)
 			if err != nil {
 				r.setCompliance(policy, policyv1.NonCompliant)
-				OpLog.Error(err, "Could not handle existing musthave object")
+				OpLog.Error(err, "Could not handle missing musthave object")
 
-				return err
+				err = r.DynamicWatcher.EndQueryBatch(watcher)
+				if err != nil {
+					panic(err)
+				}
+
+				return reconcile.Result{}, err
 			}
 
 			r.setCompliance(policy, policyv1.Compliant)
 
-			return nil
-		}
+			err = r.DynamicWatcher.EndQueryBatch(watcher)
+			if err != nil {
+				panic(err)
+			}
 
-		// Inform
+			return reconcile.Result{Requeue: true}, nil
+		}
+		// inform only
 		r.setCompliance(policy, policyv1.NonCompliant)
-
-		return nil
-	}
-
-	// Object does not exist and it should not exist, emit success event
-	if !exists && !shouldExist {
-		OpLog.Info("The object does not exist and is compliant with the mustnothave compliance type")
-		// Future implementation: Possibly emit a success event
-
-		return nil
 	}
 
-	// Object exists, now need to validate field to make sure they match
-	if exists {
-		OpLog.Info("The object already exists. Checking fields to verify matching specs")
-		// Future implementation: Verify the specs of the object matches the one on the cluster
+	r.setCompliance(policy, policyv1.Compliant)
```

Comment:
I think this is so that it could be marked as Compliant in inform mode, but actually if `createPolicyResources` is called in inform mode, then I think it should be marked as NonCompliant (which something above already does). tl;dr I think this line should be removed.